subsys/bluetooth/common/Kconfig

```
@@ -183,12 +191,44 @@ config BT_BUF_CMD_TX_SIZE
 	  to the maximum of 255.
 
 config BT_BUF_CMD_TX_COUNT
-	int "Number of HCI command buffers"
-	default 2
-	range 2 64
+	int "Number of HCI command buffers" if !BT_HCI_RAW || !HAS_BT_CTLR
+	# !BT_HCI_ACL_FLOW_CONTROL ==> Permit configurable Num_HCI_Command_Packets for
+	# Host-only build but hidden when HAS_BT_CTLR=y (default 1 for Controller-only build).
+	# !(BT_HCI_RAW && HAS_BT_CTLR) ==> Host-only build.
+	depends on !BT_HCI_RAW || !HAS_BT_CTLR || !BT_HCI_ACL_FLOW_CONTROL
+	# BT_MESH is starting Extended Advertising in the callback run in the system work queue and
+	# as the Host also uses the system work queue for Tx towards the Controller, one additional
+	# HCI command buffer is needed.
+	range 2 64 if BT_MESH
```

This is not fully correct any more because mesh now by default uses a separate workq:
https://github.com/zephyrproject-rtos/zephyr/blob/e885c8c42f775f92176e7ad7eb71d8dcdc2bb4e0/subsys/bluetooth/mesh/Kconfig#L106-L135

I suggest to set limit range for `BT_MESH_WORKQ_SYS`:
```suggestion
	# BT_MESH when compiled with BT_MESH_WORKQ_SYS is starting Extended Advertising in the callback
	# run in the system work queue and as the Host also uses the system work queue for Tx towards
	# the Controller, one additional HCI command buffer is needed.
	range 2 64 if BT_MESH_WORKQ_SYS
```